lib/inets/src/http_server/httpd_file.erl

```diff
@@ -35,13 +35,17 @@ handle_error(enotdir, Op, ModData, Path) ->
 	         ": A component of the file name is not a directory");
 handle_error(eisdir, Op, ModData, Path) ->
     handle_error(403, Op, ModData, Path,
-	         ":Illegal operation expected a file not a directory");
+	         ": Illegal operation expected a file not a directory");
+handle_error(enametoolong, Op, ModData, Path) ->
+    handle_error(422, Op, ModData, Path,
+	         ": Filename too long");
 handle_error(emfile, Op, _ModData, Path) ->
-    handle_error(500, Op, none, Path, ": Too many open files");
+    handle_error(422, Op, none, Path, ": Too many open files");
```

Comment:
I might be mistaken but I think this should be a 503.

My understanding is that 4xx errors mean that the client shouldn't repeat the request because it is most probably a client error while 5xx errors typically mean: try again later.

Reply:
Well, ok I agree we can make this one 503!